### PR TITLE
Improved robustness to instable Busses

### PR DESCRIPTION
**Pull request by henfri**
Previously, the Plugin disconnected from OWServer when a problem occured.
Now it tries to complete the sensor-cycle.

### CODE REVIEW

**Comment by ohinckel** [comment]
Is this patch is well tested? I don't use onewire so I can't test it. Maybe others here can?

File: onewire/__init__.py [hunk 1]
```diff
@@ -388,23 +390,28 @@ def _sensor_cycle(self):
                     value = self.read('/uncached' + path).decode()
                     if key.startswith('T') and value == '85.0000':
                         self.logger.info("1-Wire: problem reading {0}. Wiring problem?".format(addr))
+                        temperror=True
```
Comment (bmxp) [inline onewire/__init__.py]: ``temperror`` braucht es IMHO hier gar nicht: wenn der Temperaturfehler auftritt, dann wird mit dem nachfolgenden ``continue`` in der for-loop das nächste Element gewählt. Daher braucht auch weiter unten nicht auf temperror getestet werden.

**Comment by henfri** [comment]
Hello,

for me this patch improved the reliability significanlty and it is running stable now (opposed to before).
We could also make this modification as optional (controlled by a parameter) for a while.

@bmxp you are right.

Greetings,
Hendrik

**Comment by bmxp** [comment]
I have no onewire devices attached to my SHNG, I can not test it but I think the changes should work.

**Comment by maddinador** [comment]
I can collect my ow parts and set up a test env (2 busses, 5 devices) for this patch this week, if needed. 
But I didn't had any problems last time (about a year ago), so I cannot tell whether this PR would make it better...

File: onewire/__init__.py [hunk 2]
```diff
@@ -377,6 +378,7 @@ def _sensor_cycle(self):
         start = time.time()
         for addr in self._sensors:
             if not self.alive:
+                self.logger.info("1-Wire: Self not alive".format(addr))
```
[the remark this answers is not in the document]
Comment (henfri) [reply onewire/__init__.py]: done. Thanks for the suggestion

**Comment by henfri** [comment]
Thanks for your suggestions.
I have implemented them. I hope this is visible?

**Comment by maddinador** [comment]
Yeah, ok from my side! :+1:

**Pull request merged**